concordium-consensus/src/Concordium/GlobalState/Persistent/LMDB.hs

```diff
@@ -607,16 +609,19 @@ resizeOnFull :: (MonadLogger m, MonadIO m, MonadState s m, HasDatabaseHandlers p
   -> m a
 resizeOnFull addSize a = do
     dbh <- use dbHandlers
-    inner dbh
+    resizeOnFullInternal addSize dbh a
+
+resizeOnFullInternal :: (MonadIO m, MonadLogger m) => Int -> DatabaseHandlers pv st -> (DatabaseHandlers pv st -> IO b) -> m b
```

Comment:
There is a `resizeOnResizedInternal` function which is different from `resizeOnResized` in that it takes the database handlers as a parameter, instead of reading them from the monadic state.

Since we are performing recovery during Skov initialization, we're not working in a monad that `HasDatabaseHandlers`. Using `resizeOnFullInternal` here is similar to using `resizeOnResizedInternal` (through `openReadOnlyDatabase`) in the database exporter (which also doesn't rely on a Skov state).

Reply:
But can you please add documentation?